x/gov/client/cli/query.go

```diff
@@ -24,7 +29,905 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
 		RunE:                       client.ValidateCmd,
 	}
 
-	// this line is used by starport scaffolding # 1
+	cmd.AddCommand(
+		GetCmdQueryProposal(),
+		GetCmdQueryProposals(),
+		GetCmdQueryVote(),
+		GetCmdQueryVotes(),
+		GetCmdQueryParam(),
+		GetCmdQueryParams(),
+		GetCmdQueryProposer(),
+		GetCmdQueryDeposit(),
+		GetCmdQueryDeposits(),
+		GetCmdQueryTally(),
+	)
 
 	return cmd
 }
+
+// GetCmdQueryProposal implements the query proposal command.
+func GetCmdQueryProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "proposal [proposal-id]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Query details of a single proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Query details for a proposal. You can find the
+proposal-id by running "%s query gov proposals".
+Example:
+$ %s query gov proposal 1
+`,
+				version.AppName, version.AppName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			// validate that the proposal id is a uint
+			proposalID, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("proposal-id %s not a valid uint, please input a valid proposal-id", args[0])
+			}
+
+			// Query the proposal
+			res, err := queryClient.Proposal(
+				context.Background(),
+				&types.QueryProposalRequest{ProposalId: proposalID},
+			)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintOutput(&res.Proposal)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdQueryProposals implements a query proposals command. Command to Get a
+// Proposal Information.
+func GetCmdQueryProposals() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "proposals",
+		Short: "Query proposals with optional filters",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Query for a all paginated proposals that match optional filters:
+Example:
+$ %s query gov proposals --depositor 2
+$ %s query gov proposals --voter 2
+$ %s query gov proposals --status (DepositPeriod|VotingPeriod|Passed|Rejected)
+`,
+				version.AppName, version.AppName, version.AppName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			depositorID, _ := cmd.Flags().GetUint64(flagDepositor)
+			voterID, _ := cmd.Flags().GetUint64(flagVoter)
+			strProposalStatus, _ := cmd.Flags().GetString(flagStatus)
+			numLimit, _ := cmd.Flags().GetUint64(flagNumLimit)
+
+			var proposalStatus types.ProposalStatus
+
+			if len(strProposalStatus) != 0 {
+				proposalStatus1, err := types.ProposalStatusFromString(NormalizeProposalStatus(strProposalStatus))
```

Comment:
I don't think using numbers in variable names is the best practice. @anilCSE what do you say?